core/src/test/scala/unit/kafka/admin/ReassignPartitionsServiceClientTest.scala

```diff
@@ -0,0 +1,394 @@
+/*
```

Comment:
I like this test class! Thanks for adding it

One suggestion I have is to prefer smaller test methods that have descriptive names with regards on what they test. This helps narrow down failures as fast as possible (ideally, the method name should tell you). This was highly recommended in [Growing Object-Oriented Software, Guided by Tests (Beck Signature)](https://www.amazon.co.uk/Growing-Object-Oriented-Software-Guided-Signature/dp/0321503627) which I found was a very good book

To give a concrete example: I'd change the `testGetReplicaAssignmentForTopics` method to be 3 separate tests:
`testGetReplicaAssignmentForTopicsShouldFetchPartitionsForAllKnownTopics`
`testGetReplicaAssignmentForTopicsShouldNotFetchAllAvailableTopics`
`testGetReplicaAssignmentForTopicsShouldThrowExceptionOnUnknownTopic`

This results in long test method names but that's a feature IMO - people skimming over the test quickly realize what's tested vs what's not - and test failures suddenly become very clear (in the IDE or CI), since the test method clearly communicates what's failed